internal/pkg/service/stream/definition/repository/sink/sink_enable_test.go

```diff
@@ -0,0 +1,256 @@
+package sink_test
+
+import (
+	"context"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/keboola/go-client/pkg/keboola"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	serviceErrors "github.com/keboola/keboola-as-code/internal/pkg/service/common/errors"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/utctime"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/definition"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/definition/key"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/dependencies"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/test"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/etcdhelper"
+)
+
+func TestSinkRepository_Enable(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	now := utctime.MustParse("2000-01-01T01:00:00.000Z").Time()
+	by := test.ByUser()
+
+	d, mocked := dependencies.NewMockedServiceScope(t)
+	client := mocked.TestEtcdClient()
+	repo := d.DefinitionRepository().Sink()
+	ignoredEtcdKeys := etcdhelper.WithIgnoredKeyPattern("^(definition/branch|definition/source)")
+
+	// Fixtures
+	projectID := keboola.ProjectID(123)
+	branchKey := key.BranchKey{ProjectID: projectID, BranchID: 567}
+	sourceKey := key.SourceKey{BranchKey: branchKey, SourceID: "my-source"}
+	sinkKey := key.SinkKey{SourceKey: sourceKey, SinkID: "my-sink"}
+
+	// Create branch - ok
+	// -----------------------------------------------------------------------------------------------------------------
+	{
+		branch := test.NewBranch(branchKey)
+		require.NoError(t, d.DefinitionRepository().Branch().Create(&branch, now, by).Do(ctx).Err())
+	}
+
+	// Enable - not found
+	// -----------------------------------------------------------------------------------------------------------------
+	if err := repo.Enable(sinkKey, now, by).Do(ctx).Err(); assert.Error(t, err) {
+		assert.Equal(t, `sink "my-sink" not found in the source`, err.Error())
+		serviceErrors.AssertErrorStatusCode(t, http.StatusNotFound, err)
+	}
```

Comment:
Not in code block. Mentioning to reduce your time to fix these cases.

Reply:
https://github.com/keboola/keboola-as-code/pull/1781/commits/5c3a798ab8c52bc68d974484f69f5ed1929fefe5